client/src/utils/blockstackHelpers.js

```diff
@@ -0,0 +1,18 @@
+const postObservation = async (userSession, observation, observationNumber) => {
+  const fileNumber = `${observationNumber}`.padStart(7, '0');
+  userSession.putFile(`observation/${fileNumber}.json`, JSON.stringify(observation));
+  const result = await userSession.putFile(`observations.json`, JSON.stringify(observation));
+  return result;
+};
```

Comment:
This function is a bit unclear. I understand it, but I think it should cleared up.

- The second argument, `observation`, is really expecting an array of observations not just one
- The first `putFile` call should be putting an individual observation, not an array

Maybe separate putting an individual observation and the list of observations into different functions. Up to you.

Reply:
_The second argument, observation, is really expecting an array of observations not just one_

No, the second argument is definitely only ever expecting one observation.  This function receives 1) The user session, 2) the form data from the survey, 3) the number of the next observation/00000x.json file, derived from the length of the array we fetched from observations.json

Reply:
That's a good catch.  I wrote the whole observations.json and added the sort of backup post to create the old-style observations/00000x.json after.  You're right, I'm posting the whole array in both cases.  I will fix that.